.github/workflows/unit-tests.yml

```diff
@@ -23,21 +23,21 @@ jobs:
       matrix:
         include:
           # MacOSX
-          - { compiler: clang,    os: macos-12,     buildType: Debug, cxxStandard: 14 }
+          - { compiler: clang,    os: macos-12,     buildType: Debug }
```

Comment:
IMO it makes sense to fix/set the Boost version here as it is not arbitrary but depends on the choice of macos version. See also the debug output of https://github.com/Return-To-The-Roots/s25client/pull/1580/files#diff-cd0effce06f425599e96952f4d5f684641a683f46d9909e7c74a9f8b2f03ccd3R113-R114 which may help when something changes. So I'd include that (debug output and explicit version here)

Reply:
I wrote it down now in the comments